src/authn/auth0.js

```diff
@@ -0,0 +1,61 @@
+import express from 'express'
+import Auth0Strategy from 'passport-auth0'
+import passport from 'passport'
+import assert from 'assert'
+import User from './../user'
+import url from 'url';
+
+class Auth0Login {
+  constructor(options) {
+    assert(options, 'options are required');
+    assert(options.cfg, 'options.cfg is required');
+    assert(options.cfg.auth0, 'options.cfg.auth0 is required');
+    assert(options.cfg.auth0.domain, 'options.cfg.auth0.domain is required');
+    assert(options.cfg.auth0.clientId, 'options.cfg.auth0.clientId is required');
+    assert(options.cfg.auth0.clientSecret, 'options.cfg.auth0.clientSecret is required');
+    assert(options.authorize, 'options.authorize is required');
+
+    this.cfg = options.cfg;
+    this.authorize = options.authorize;
+
+    passport.use(new Auth0Strategy({
+      domain: options.cfg.auth0.domain,
+      clientID: options.cfg.auth0.clientId,
+      clientSecret: options.cfg.auth0.clientSecret,
+      callbackURL: url.resolve(options.cfg.server.publicUrl, '/auth0/login'),
+    }, this.auth0Callback.bind(this)));
+  }
+
+  router() {
+    let router = new express.Router();
+
+    console.log('setting up auth0/login');
+    router.get('/login', passport.authenticate('auth0', {
+      failureRedirect: '/',
+      failureFlash: true
+    }), (req, res) => {
+      res.redirect('/');
+      return;
+    });
+
+    return router;
+  };
+
+  // for the moment, we *only* support passwordless login with Auth0.
+  auth0Callback(accessToken, refreshToken, extraParams, profile, done) {
+    try {
+      let user = new User();
+      console.log(profile._json);
+      if (!profile._json.email_verified) {
+        throw new Error('email is not verified');
+      }
+      user.identity = 'passwordless/' + profile._json.email;
```

Comment:
There's no such thing as an "auth0 account".  At least, we won't be using such a thing.  When the IAM team has everything hooked up, there will be *mozilla* accounts (available to all mozillians), and also the option to use social logins (probably github and google) and even to link those.  But in no case is user authentication information stored with auth0 directly.  I'm still vague on how that will work, so I'm not really worrying about it yet.

I suspect that when we get to that point, we'll need to have multiple notions of identity -- some github-associated, some ldap-associated, etc.  But at this point that future is very hazy so I'm going to wait until it's clearer.

But I agree that `email/<email>` would be a better choice here, since it suggests "this person has access to this email account" and no more.